internal/service/dynamodb/find.go

```diff
@@ -119,3 +119,20 @@ func FindDynamoDBTTLRDescriptionByTableName(conn *dynamodb.DynamoDB, tableName s
 
 	return output.TimeToLiveDescription, nil
 }
+
+func FindContributorInsights(ctx context.Context, conn *dynamodb.DynamoDB, tableName, indexName string) (*dynamodb.DescribeContributorInsightsOutput, error) {
+	input := &dynamodb.DescribeContributorInsightsInput{
+		TableName: aws.String(tableName),
+	}
+
+	if indexName != "" {
+		input.IndexName = aws.String(indexName)
+	}
+
+	output, err := conn.DescribeContributorInsightsWithContext(ctx, input)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Treat disabled as not found.

```suggestion
	output, err := conn.DescribeContributorInsightsWithContext(ctx, input)

        if tfawserr.ErrCodeEquals(err, dynamodb.ErrCodeResourceNotFoundException) {
		return nil, &resource.NotFoundError{
			LastError:   err,
			LastRequest: input,
		}
        }
        
	if err != nil {
		return nil, err
	}
	
	if status := aws.StringValue(output. ContributorInsightsStatus); status == dynamodb.ContributorInsightsStatusDisabled {
		return nil, &resource.NotFoundError{
			Message:     state,
			LastRequest: input,
		}
	}
	
	if output == nil {
		return nil, tfresource.NewEmptyResultError(input)
	}
```